app/models/async_transaction/base.rb

```diff
@@ -25,6 +26,14 @@ def parsed_metadata
     validates :transaction_id,
               uniqueness: { scope: :source, message: 'Transaction ID must be unique within a source.' }
 
+    def delete_stale
+      Base.where(
+        'created_at < ?', DELETE_COMPLETED_AFTER.ago
+      ).where(
+        status: Base::COMPLETED
+      ).find_each(&:destroy!)
+    end
+
```

Comment:
Perhaps align the queries in a 'one query per line' fashion, for readability?  For example:

```ruby
def delete_stale
  Base
    .where('created_at < ?', DELETE_COMPLETED_AFTER.ago)
    .where(status: Base::COMPLETED)
    .find_each(&:destroy!)
end
```

Reply:
✅